File: lec13/temp.c
#include <stdio.h>
#include <stdlib.h>
#define LENGTH 1000000

double convert(double f) {
  return (f - 32)* 5/9;
}


int main(void) {
  double temps[LENGTH];
  
  for(int i = 0; i < LENGTH; i++) {
    temps[i] = ((double) rand()) / RAND_MAX  * 150 - 50;
  }

  for(int i = 0; i < LENGTH; i++) {
    temps[i] = convert(temps[i]);
  }

  printf("[");
  for(int i = 0; i < LENGTH; i++) {
    printf("%f,",temps[i]);
  }
  printf("]\n");

  return 0;
}

File: lec13/temp.py
import random

def convert(f):
    return (f - 32) * 5/9 


temps = []
for i in range(100000000):
    temps.append(random.random() * 150 - 50)

for i in range(len(temps)):
    temps[i] = convert(temps[i])


print(temps)



    
